###############################################################################
#
# Copyright (c) 2009, Oracle and/or its affiliates. All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
#
###############################################################################

Set default Xsession scripts/programs for Solaris/OpenSolaris.

--- a/config/Xsession.in	2020-09-22 08:10:04.022219142 -0700
+++ b/config/Xsession.in	2020-09-22 08:14:31.762399514 -0700
@@ -47,6 +47,14 @@
 	esac
 esac
 
+XCOMM Run system session script if present - these will do various setup,
+XCOMM then run $HOME/.xsession if it exists, so we can just run them
+if [[ -x /etc/X11/gdm/Xsession && -x /usr/bin/dtstart ]] ; then
+        exec /etc/X11/gdm/Xsession /usr/bin/dtstart jds
+elif [[ -x /usr/dt/config/Xsession.jds ]] ; then
+        exec /usr/dt/config/Xsession.jds
+fi
+
 XCOMM The startup script is not intended to have arguments.
 
 startup=$HOME/.xsession
@@ -62,5 +70,6 @@
 	if [ -r "$resources" ]; then
 		BINDIR/xrdb -load "$resources"
 	fi
-	exec BINDIR/xsm
+        BINDIR/xterm -geometry 80x24+10+10 -ls &
+        exec BINDIR/twm
 fi
